### DISCUSSION TITLE
Search Refiners Checkbox shows unchecked once you click on apply version 3.15

### DISCUSSION THREAD

**Discussion by tusharprithviraj** [microsoft-search/pnp-modern-search#2724] [Q&A]
Hi Team,

I am working on Search Webparts verson 3.15.

For search refiners i am facing one issue, i have multiple refiners job title, city, department, classification. These all are set as multiple value selection.

The refiners works fine but some combination when you have 3 different refiners selected and click on apply  it filters the results but the checkbox checkmarked  is gone.

i tried debugging the code the newfilters.length in checkboxtempate.tsx file is setting as zero even if it has values in there.

Also some refiners have duplicate values such as USA and usa for the uppercase it works fine but for lowercase it filters the data and the checkmarked is gone.

Any help would be appreciated, please find the attached screenshot for the same.
![image](https://user-images.githubusercontent.com/46158835/215522392-7604ff64-6d24-41ba-aeb5-b4d4ca913177.png)

**Comment by wobba** [comment]
I suggest trying v4 instead to see if the issue is the same or not. Unless someone have time to investigate v3. As for lowercase+uppercase values this is how it's indexed and returned on the API.

**Reply by tusharprithviraj** [reply]
Hi Wobba ...yes in V4 it's working fine.

**Reply by wobba** [reply]
I have no plans to make more updates to v3. So if it's not working in v3.23 which is the latest release then v4 is what you should focus on.
